project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_visitors/c4m_og_invite_visitors.module

```diff
@@ -0,0 +1,395 @@
+
+  // For private or moderated groups, return false. Else check for member.
+  if ($access_type['type'] == 'private' || $membership_request_value == 'moderated') {
+    return FALSE;
+  }
+  else {
+    return og_is_member($group->type, $group->nid, 'user', $user);
+  }
+}
+
+/**
+ * Implements hook_og_permission().
+ */
+function c4m_og_invite_visitors_og_permission() {
+  $items = array();
+  $items['invite members'] = array(
+    'title' => t('Invite members'),
+    'description' => t('Users may invite other users to the group without approval.'),
+    'default role' => array(OG_ADMINISTRATOR_ROLE),
+  );
+
+  return $items;
+}
+
+/**
+ * Main OG Invite members form.
+ */
+function _c4m_og_invite_visitors_new_users_form($form, &$form_state, $group_type, $gid) {
+  og_set_breadcrumb($group_type, $gid, array(l(t('Group'), "$group_type/$gid/group")));
+  $group = entity_load_single($group_type, $gid);
+  $label = entity_label($group_type, $group);
+  list(, , $bundle) = entity_extract_ids($group_type, $group);
+  $og_roles = array('') + og_roles($group_type, $bundle, $gid, FALSE, FALSE);
+
+  $form['group_type'] = array('#type' => 'value', '#value' => $group_type);
+  $form['gid'] = array('#type' => 'value', '#value' => $gid);
+
+  $form['c4m_og_invite_visitors'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Invite new users to %group', array('%group' => $label)),
+  );
+  $form['c4m_og_invite_visitors']['invitee'] = array(
+    '#type' => 'textarea',
+    '#title' => t('User email(s)'),
+    '#description' => t("Use commas or new lines to split email addresses. The new users will be created and invited as members of this group. A one time login link will be sent to the invitees\' email."),
+    '#required' => TRUE,
+  );
+  $form['og_invite_people']['roles'] = array(
+    '#type' => 'select',
+    '#multiple' => TRUE,
+    '#title' => t('Assign roles'),
+    '#description' => t('Choose one or more roles you would like to assign to the users. ("member" role is assigned by default)'),
+    '#options' => $og_roles,
+    '#size' => 5,
+  );
+  $form['c4m_og_invite_visitors']['state'] = array(
+    '#type' => 'select',
+    '#title' => t('Membership state'),
+    '#options' => og_group_content_states(),
+    '#description' => t('Select the state of the membership.'),
+  );
+  $form['c4m_og_invite_visitors']['personal_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#description' => t('Add a personal subject to the invitation email.'),
+  );
+  $form['c4m_og_invite_visitors']['personal_message'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Add a message (optional)'),
+    '#description' => t('Add a personal message to the invitation email.'),
+  );
+
+  $field_names = og_get_group_audience_fields('user', 'user', $group_type);
+  $field_name = !empty($form_state['values']['field_name']) ? $form_state['values']['field_name'] : key($field_names);
+
+  if (count($field_names) > 1) {
```

Comment:
After consulting with the maintainer of OG, I will remove this condition. There is no reason to check for multiple group audience fields here.